File: android/Ninehoto.Android/Models/MediaItem.cs
using System;

namespace Ninehoto.Android.Models
{
    public class MediaItem
    {
        public string Id { get; set; } = "";
        public MediaType MediaType { get; set; }
        public DateTime? CreationDate { get; set; }
        public int PixelWidth { get; set; }
        public int PixelHeight { get; set; }
        public double Duration { get; set; }
        public bool IsVideo { get; set; }

        public enum MediaType
        {
            Photo,
            Video,
            Unknown
        }

        public double AspectRatio => PixelHeight > 0 ? (double)PixelWidth / PixelHeight : 1;
        public bool IsPortrait => AspectRatio < 1;
        public bool IsLandscape => AspectRatio > 1;
        public bool IsSquare => Math.Abs(AspectRatio - 1) < 0.01;

        public string FormattedDuration
        {
            get
            {
                if (!IsVideo || Duration <= 0)
                    return "";
                
                var minutes = (int)Duration / 60;
                var seconds = (int)Duration % 60;
                return $"{minutes}:{seconds:D2}";
            }
        }
    }

    public class MediaRepository
    {
        private readonly IMediaService _mediaService;

        public MediaRepository(IMediaService mediaService)
        {
            _mediaService = mediaService;
        }

        public async System.Threading.Tasks.Task<MediaStoreResult> RequestPermissionAsync()
        {
            return await _mediaService.RequestPermissionAsync();
        }

        public async System.Threading.Tasks.Task<System.Collections.Generic.List<MediaItem>> FetchRecentMediaAsync(int limit)
        {
            return await _mediaService.FetchRecentMediaAsync(limit);
        }

        public async System.Threading.Tasks.Task<bool> DeleteMediaAsync(System.Collections.Generic.List<string> assetIds)
        {
            return await _mediaService.DeleteMediaAsync(assetIds);
        }
    }

    public class MediaStoreResult
    {
        public bool Success { get; set; }
        public string? ErrorMessage { get; set; }

        public static MediaStoreResult Ok() => new MediaStoreResult { Success = true };
        public static MediaStoreResult Fail(string message) => new MediaStoreResult { Success = false, ErrorMessage = message };
    }
}

File: ios/Ninehoto/Models/MediaItem.swift
import Foundation
import Photos

struct MediaItem: Identifiable, Equatable {
    let id: String
    let asset: PHAsset
    let mediaType: MediaType
    let creationDate: Date?
    let duration: TimeInterval?
    let fileSize: Int64?

    enum MediaType: String {
        case photo
        case video
        case unknown
    }

    init(asset: PHAsset, fileSize: Int64? = nil) {
        self.id = asset.localIdentifier
        self.asset = asset
        self.fileSize = fileSize

        switch asset.mediaType {
        case .image:
            self.mediaType = .photo
        case .video:
            self.mediaType = .video
        default:
            self.mediaType = .unknown
        }

        self.creationDate = asset.creationDate
        self.duration = asset.duration
    }

    var isVideo: Bool {
        mediaType == .video
    }

    var formattedDuration: String? {
        guard let duration = duration else { return nil }
        let minutes = Int(duration) / 60
        let seconds = Int(duration) % 60
        return String(format: "%d:%02d", minutes, seconds)
    }

    var formattedFileSize: String? {
        guard let fileSize else { return nil }
        return fileSize.formattedAsFileSize
    }

    var aspectRatio: CGFloat {
        let width = asset.pixelWidth
        let height = asset.pixelHeight
        guard height > 0 else { return 1.0 }
        return CGFloat(width) / CGFloat(height)
    }

    var isPortrait: Bool {
        aspectRatio < 1.0
    }

    var isLandscape: Bool {
        aspectRatio > 1.0
    }

    var isSquare: Bool {
        abs(aspectRatio - 1.0) < 0.01
    }
}

extension Int64 {
    var formattedAsFileSize: String {
        if self < 1024 { return "\(self) B" }
        if self < 1024 * 1024 { return String(format: "%.1f KB", Double(self) / 1024) }
        if self < 1024 * 1024 * 1024 { return String(format: "%.1f MB", Double(self) / (1024 * 1024)) }
        return String(format: "%.1f GB", Double(self) / (1024 * 1024 * 1024))
    }
}